src/Core/src/Handlers/WebView/WebViewHandler.Windows.cs

```diff
@@ -350,9 +354,10 @@ void OnCoreWebView2Initialized(WebView2 sender, CoreWebView2InitializedEventArgs
 				sender.CoreWebView2.NavigationStarting += OnNavigationStarting;
 				sender.CoreWebView2.NavigationCompleted += OnNavigationCompleted;
 
-				if (Handler is WebViewHandler handler)
+				if (Handler is WebViewHandler handler && handler.VirtualView is not null)
 				{
 					sender.UpdateUserAgent(handler.VirtualView);
+					handler.SyncPlatformCookies(sender.CoreWebView2.Source).FireAndForget();
```

Comment:
I don't think the source will be set to the right value at this point

![image](https://github.com/user-attachments/assets/f505d240-9079-4e8f-926c-1d884f5c3027)

For the UItest can you add a part that validates the cookie was set?

like maybe after navigated has fired you can read from the cookies to see if the cookie you added is present?

Reply:
We've added cookie validation to the UI test as per your suggestion. Could you please verify once?